api/reviews_test.go

```diff
@@ -0,0 +1,185 @@
+package api
+
+import (
+	"context"
+	"github.com/cadmiumcat/books-api/apierrors"
+	"github.com/cadmiumcat/books-api/interfaces/mock"
+	"github.com/cadmiumcat/books-api/models"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+const (
+	bookID   = "1"
+	reviewID = "123"
+	emptyID  = ""
+)
+
+var errMongoDB = errors.New("unexpected error in MongoDB")
+
+func TestReviewsEndpoints(t *testing.T) {
```

Comment:
Should this function just be testing the ReviewEndpoint? `/books/<book_id>/reviews/<review_id>`

You can have a separate function to deal with the reviews endpoint `/books/<book_id/reviews`